### PR TITLE
jrickard-vmware IPI 

### PR DESCRIPTION
**Pull request by day0hero**
Installation walkthrough of OCP 4.7 IPI on vmware. 
@itewk

### CODE REVIEW

**Comment by itewk** [comment]
@claudiol @ckupe can you all review this with @jflowers ?

**Comment by itewk** [comment]
@claudiol @ckupe  @jflowers ping

**Review by jflowers — CHANGES_REQUESTED** [review]
Can we get the same major sections and format to match the other docs created so far?
For example:
## Overview
## VMWare Configuration Requirements
### Example VMWare Configuration
## Installing OpenShift
### Bundling Content and Moving it to the Disconnected Environment
### Run OpenShift Install


Also it is nice to indicate what machine commands are being run from. In the other docs this has been noted as a comment in the first line of a code block. For example:
https://github.com/redhat-cop/ocp-disconnected-docs/blob/main/IPIonMAGInstall.md#2-push-bundle-to-registry

**Comment by itewk** [comment]
@jrickard got time to iterate on this PR with @jflowers ?

File: IPIonVMWare.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
+# Installing OpenShift in vmWare Disconnected using IPI
```
[the remark this answers is not in the document]
Comment (day0hero) [reply IPIonVMWare.md:1]: i just looked on their site - its VMware; I will fix this

File: IPIonVMWare.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+- RHEL 8.2 Template 
+
+### Prerequisites
+- at least one internet facing machine
+- dns server with zones and subdomains
+- dns entries (A Records) for registry, api and wildcard apps `*.apps`
+- dhcp service
+
+### vmWare Role Permissions
+[vmware permissions](https://docs.openshift.com/container-platform/4.6/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html#installation-vsphere-installer-infra-requirements_installing-vsphere-installer-provisioned-customizations)
```
[the remark this answers is not in the document]
Comment (day0hero) [reply IPIonVMWare.md:22]: done.

File: IPIonVMWare.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+1. Create a new role:
+- Click on menu, then click administration
+- Click on `Roles` under `Access Control`
+- Click the `+` sign to create a new role
+- Using the document above, assign the correct permissions to the role
```
[the remark this answers is not in the document]
Comment (day0hero) [reply IPIonVMWare.md:28]: The documentation doesn't go through creating the role and applying the permissions. If one has experience creating roles and permissions in vcenter this is probably trivial, but if not, at least to me, it wasn't straightforward

File: IPIonVMWare.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+openshift-client-linux.tar.gz \
+openshift-install-linux.tar.gz \
+opm-linux.tar.gz \
+rhcos-vmware.x86_64.ova
+<your_pull_secret>.json 
+```
+
+8. SCP the file (or use whatever transfer means available) to the disconnected node:
+```
+scp mirror_bundle.tar <ip_of_disconnected_node>:~
```
[the remark this answers is not in the document]
Comment (day0hero) [reply IPIonVMWare.md:106]: Line 111: 
<p class="callout info"> A virtual machine is deployed from the RHEL8 template and is configured with only the internal portgroups.</p>

Do I need to walk through creating a vm using a template? I can do it, but my template names will be different than what they are using...

File: IPIonVMWare.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+
+### Preparing the mirror node
+
+<p class="callout info"> A virtual machine is deployed from the RHEL8 template and is configured with both the public and internal portgroups.</p>
+
+1. Download the quay pull secret from [cloud.redhat.com](https://cloud.redhat.com)
+
+2. Download the OpenShift commadline utilities and OVA files
+```
+curl -LfO  https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.7/latest/rhcos-vmware.x86_64.ova
```
[the remark this answers is not in the document]
Comment (day0hero) [reply IPIonVMWare.md:53]: LMK what you want to do with this - my intention was to make it as vanilla (follow the docs as much as possible)

File: IPIonVMWare.md [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+|jq| view json in readable format | no (will make managing json much easier|
+
+3. Install packages:
+```
+dnf install -y httpd jq unzip
+```
+
+4. Install the vcenter certificates
+```
+curl -k -LfO https://vcenter.fq.dn/certs/download.zip`
```
[the remark this answers is not in the document]
Comment (day0hero) [reply IPIonVMWare.md:134]: changed fq.dn to example.com

**Comment by itewk** [comment]
my two cents. merge it as is. iterate out the rest later with smaller PRs and such. this is far better then anything else out there (or not out there).

**Comment by itewk** [comment]
@jflowers can we merge this?

**Comment by jflowers** [comment]
> @jflowers can we merge this?

I guess so. We have already found that we are getting calls on parts that we didn't explain well. As long as people are ready to field the questions/support go for it

**Comment by day0hero** [comment]
@

> > @jflowers can we merge this?
> 
> I guess so. We have already found that we are getting calls on parts that we didn't explain well. As long as people are ready to field the questions/support go for it

What are the parts that aren't explained well and I will make the corrections

**Comment by itewk** [comment]
@day0hero @jflowers how about, lets merge, cuz this in its current state is better then nothing/anything else. and then can do follow on prs to make better as people use and find enahcments. perfect / enemy / good and all that.

**Comment by jflowers** [comment]
> @day0hero @jflowers how about, lets merge, cuz this in its current state is better then nothing/anything else. and then can do follow on prs to make better as people use and find enahcments. perfect / enemy / good and all that.

Go for it @itewk

**Pull request merged**